resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -433,12 +442,15 @@ private[spark] class KubernetesClusterSchedulerBackend(
     rpcEnv: RpcEnv,
     sparkProperties: Seq[(String, String)])
     extends DriverEndpoint(rpcEnv, sparkProperties) {
-    private val externalShufflePort = conf.getInt("spark.shuffle.service.port", 7337)
 
     override def onDisconnected(rpcAddress: RpcAddress): Unit = {
       addressToExecutorId.get(rpcAddress).foreach { executorId =>
         if (disableExecutor(executorId)) {
-            executorsToRemove.add(executorId)
+          RUNNING_EXECUTOR_PODS_LOCK.synchronized {
+            runningExecutorsToPods.get(executorId).foreach { pod =>
+              disconnectedPodsByExecutorIdPendingRemoval(executorId) = pod
```

Comment:
can you explain a bit why it's safe to remove the executor directly here, rather than going through the `executorsToRemove` set first?

Reply:
We don't remove the executor directly here. Some of the logic has changed and variables are renamed, so `executorsToRemove` doesn't exactly exist anymore anyways.

Reply:
Here we are logging the executor as disconnected and the allocator thread will clean it up once the exit reason is known. I believe this is the same semantics as before.

Reply:
Yup. No change in semantics just that the marking is done after verifying that the executor is still running/tracked.